coderd/templateversions.go

```diff
@@ -1582,10 +1583,63 @@ func (api *API) postTemplateVersionsByOrganization(rw http.ResponseWriter, r *ht
 		}
 	}
 
+	var files fs.FS
```

Comment:
This was previously only necessary for dynamic parameters, but we now need it in both dynamic and classic templates because we validate presets in both cases. It feels a tad wasteful if considered in isolation, but I'd like to update the classic template case that uses `tfparse` to use this `fs.FS` as well. That would be a small first step in getting rid of tfparse.

Reply:
> I'd like to update the classic template case that uses tfparse to use this fs.FS as well.

There is a `tfconfig.LoadModuleFromFilesystem` but IIRC I tried using this before and ran into issues. 

> Maybe we should only do preset validation if using dynamic parameters.

@Emyrk what potential issues do we avoid by doing this? Tying a behaviour relating to a 'GA' feature to a per-template 'Beta' feature feels like it would be unexpected.

Reply:
@johnstcn `preview` is a large part of the dynamic parameters feature. It just feels premature to push `preview` to GA for this parsing, while the rest of its use is still getting tested in Beta.

If there is a bug in `preview`, we will know when Beta is in the wild

Reply:
As discussed internally, preset validation has been moved to dynamic parameters only and will graduate to general use alongside dynamic parameters